### ISSUE TITLE
hwdb.d in volatile /run directory?

### ISSUE THREAD

**Issue by whot** [systemd/systemd#127]
Opening on request of Lennart, see https://github.com/systemd/systemd/pull/115#issuecomment-110320638

The question is whether the hwdb should read from /run/udev/hwdb.d or not, given that dropping any hwdb file in there requires re-building the database. This was documented in man/hwdb.xml but the implementation was missing until the pull request linked above.

Use-case is here:
http://lists.freedesktop.org/archives/wayland-devel/2015-June/022476.html
specifically: libinput ships a hwdb.d and a rules file to add properties for certain devices. To run the tests suite, these need to be put in place before a test (with a rules reload and a hwdb update) and removed after the test again.

This currently only affects the test suite where we shouldn't rely on libinput being installed, let alone having the latest hwdb updates. So we need some temporary directory for hwdb files, but personally I don't care whether it's /run or can be provided as an argument to hwdb update.

**Comment by dvdhrm** [comment]
If you place your hwdb files into /run, you still have to trigger a rebuild. This rebuild will still place the binary hwdb into /etc. Hence, your testing data will still be globally installed. This will even stay there once you remove your data from /run again. Only if you then rebuild again (which is not automatically triggered) the data will get stripped.

I can see that you probably don't mind whether the data stays in the binary. You probably trigger a rebuild after your tests finished and this leak only happens if something crashed. However, I wanna make sure you're aware of that.

Long story short: Supporting /run sounds good to me, as long as you're aware of the side-effects.

**Comment by whot** [comment]
Yeah, I'm aware of that but tbh I don't (need to) care too much where the binary database ends up (and whether it leaks, we clean up and rebuild before every test anyway). That's system configuration for me and since the tests need to be run as root anyway everything is accessible.

I'd be ok if you say "drop it in /etc" and we remove the /run references. Again, the reason for this fix was simply because /run was already documented as such but didn't do what the docs said.

As I said above, if you want to drop /run I'd also take a udevadm hwdb --update --include=/tmp/hwdb.d argument. Or just drop it in /etc, I've got options here ;)

**Comment by dvdhrm** [comment]
So if you're fine with dropping it into /etc and removing /run from the documentation, that would be best. Supporting /run in something that is inherently not about runtime configuration feels weird. 'testing' is the only use-case I can think about.

I can push a documentation fix tomorrow, if nobody objects.

**Comment by dvdhrm** [comment]
The documentation has now been fixed in 01856799a3113398da924d9b0dcc09d1e853d905.

Thanks for the report!
